src/reducers/moveThought.ts

```diff
@@ -64,389 +17,147 @@ const moveThought = (
     newPath,
     offset,
     skipRerank,
+    newRank,
```

Comment:
`Path` is now an array of ids. So we cannot pass rank. We can only access the parent entry associated to the thought id prior to the move. That's why a new rank needs to be passed here.

Reply:
What is `headRank(state, newPath)` then?

Reply:
> What is `headRank(state, newPath)` then? Is it the old rank?

Yes.  Since thoughts are now independent,  we only need `thoughtId, 'newParentId' and 'rank' to move a thought. We don't need to think about it in terms of  `oldPath` and `newPath` now.